pkg/build/builder/cmd/builder.go

```diff
@@ -188,29 +188,31 @@ func newBuilderConfigFromEnvironment(out io.Writer, needsDocker bool) (*builderC
 	return cfg, nil
 }
 
-func (c *builderConfig) setupGitEnvironment() (string, []string, error) {
+func (c *builderConfig) setupGitEnvironment() (string, []string, string, error) {
 
 	// For now, we only handle git. If not specified, we're done
 	gitSource := c.build.Spec.Source.Git
 	if gitSource == nil {
-		return "", []string{}, nil
+		return "", []string{}, "", nil
 	}
 
 	sourceSecret := c.build.Spec.Source.SourceSecret
 	gitEnv := []string{"GIT_ASKPASS=true"}
+	var outGitConfigFile string
 	// If a source secret is present, set it up and add its environment variables
 	if sourceSecret != nil {
 		// TODO: this should be refactored to let each source type manage which secrets
 		// it accepts
 		sourceURL, err := s2igit.Parse(gitSource.URI)
 		if err != nil {
-			return "", nil, fmt.Errorf("cannot parse build URL: %s", gitSource.URI)
+			return "", nil, "", fmt.Errorf("cannot parse build URL: %s", gitSource.URI)
 		}
 		scmAuths := scmauth.GitAuths(sourceURL)
 
-		secretsEnv, overrideURL, err := scmAuths.Setup(c.sourceSecretDir)
+		secretsEnv, overrideURL, gitConfigFile, err := scmAuths.Setup(c.sourceSecretDir)
+		outGitConfigFile = gitConfigFile
```

Comment:
I don't see why you are doing this, I can not seem to find anywhere else that you are using gitConfigFile anywhere in this function.

Reply:
This is passed through to `setupProxyConfig` in line 272. I'll add a godoc to make the returns clearer.

Reply:
Isn't that in a different function? Why are you copying the variable?